### ISSUE TITLE
[HL2] Half life 2 workshop doesnt load custom client.dll/server.dll

### ISSUE THREAD

**Issue by Unusuario2** [ValveSoftware/Source-1-Games#6668]
When loading a custom workshop mod that has custom client.dll/server.dll hl2 doesnt load them, making those mods useless.
If you load the mods from the command line "-game PATH/TO_THE_GAME" this is no longer a issue.

If possible fix this behavior so workshop mods can load custom client.dll/server.dll

**Comment by sylveonsylvia** [comment]
i dont think this would be a good idea, also mods built for the sdk might have incompatabilitys with the newer version of the engine hl2 complete uses

**Comment by Unusuario2** [comment]
why woundnt be a good idea? For a lot of modders this would life saver since a lot of mod use custom shaders, entities, UI and more, without a custom client/server those things cannot be achieve. Of course this would meant a new sdk or al least an update  in the SDK 2013SP. I am not saying to allow now this feature but later when an uptaded sdk or new comes out it should be a thing in half life 2 workshop

**Comment by sylveonsylvia** [comment]
with how the workshop works currently i dont think this can even be implamented, source cant hotload gamedlls

**Comment by Unusuario2** [comment]
right now source engine cannot hotload .dll files but that why this issue exist, about implementation it doesnt need to be hotloaded perse, for example when you click in the mod that you want to play, the game will restart and load those .dll on top of the original dlls, this system is currently in use by the engine to load the clients/servers for loastcoast/deathmatch/ep2 , the ideal system of course would be hotload but i prefer this rather than nothing since for modders opens a hole new world where shaders, entities, UI can be modded and not hardcoded like now

**Comment by pivotman319-owo** [comment]
_Not_ a good idea.

For one, this could possibly sacrifice engine stability as the game gets continually updated with possible breaking changes (not even the current SDK2013 MP repo is compatible with HL2 20th Anniversary as of writing), and two: **this is exactly how you introduce malware into the Steam Workshop.**

Doing this opens up a massive range of attack vectors.
